source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -259,7 +260,8 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
     if (config_->clearRouteCache() &&
         (!response->headers_to_set.empty() || !response->headers_to_append.empty() ||
          !response->headers_to_remove.empty() || !response->query_parameters_to_set.empty() ||
-         !response->query_parameters_to_remove.empty())) {
+         !response->query_parameters_to_remove.empty()) &&
+        decoder_callbacks_->downstreamCallbacks()) {
```

Comment:
After we config-validate we can add a comment here that clearRouteCache should onyl be true if  decoder_callbacks_->downstreamCallbacks() is also true

Reply:
Done.